nistats/tests/test_datasets.py

```diff
@@ -240,26 +221,24 @@ def test_make_spm_auditory_events_file():
     actual_events_data_string = replace_win_line_ends(actual_events_data_string)
     expected_events_data_string = replace_win_line_ends(expected_events_data_string)
     
-    assert_equal(actual_events_data_string, expected_events_data_string)
+    assert actual_events_data_string == expected_events_data_string
 
 
-@with_setup(setup_mock, teardown_mock)
-@with_setup(tst.setup_tmpdata, tst.teardown_tmpdata)
-def test_fetch_spm_auditory():
+def test_fetch_spm_auditory(request_mocker, tmp_path):
     import nibabel as nib
     import shutil
     saf = ["fM00223/fM00223_%03i.img" % index for index in range(4, 100)]
     saf_ = ["fM00223/fM00223_%03i.hdr" % index for index in range(4, 100)]
 
-    data_dir = os.path.join(tst.tmpdir, 'spm_auditory')
+    data_dir = str(tmp_path / 'spm_auditory')
```

Comment:
For which Python version is this syntax supported ?

Reply:
We can do this with Path objects  (which tmp_path is). It is part of the pathlib module, standard since Python 3.4 . For python2, pytest needs and installs pathlib2, an external module for the same purpose.